.github/workflows/main.yml

```
@@ -1,15 +1,55 @@
+      - name: DryRun Snakefile
+        run: docker run snakemake-tutorial snakemake --cores 1 --dryrun
+
   test:
     runs-on: ubuntu-latest
+    needs: docker-build
     steps:
-    - uses: actions/checkout@master
-    - run: docker build .
\ No newline at end of file
+      - name: Checkout Code
+        uses: actions/checkout@v2
```

actionlint (1.7.4)

the runner of "actions/checkout@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)